core/src/main/scala/dagr/core/util/ReflectiveBuilder.scala

```diff
@@ -59,41 +60,44 @@ class ReflectiveBuilder[T](val clazz: Class[T]) {
     * @param constructor a method to be examined - for now always a constructor
     */
   protected def extractParameterInformation(constructor : MethodSymbol): Unit = {
-    if (constructor.paramLists.size > 1) throw new IllegalArgumentException("Constructors with more than one parameter list are not supported.")
-    else if (constructor.paramLists.nonEmpty) {
-      // More reflection objects that are needed to extract everything
-      val hasCompanion = clazzSymbol.companion.isModule
-      val companionObject = if (hasCompanion) Some(clazzSymbol.companion.asModule) else None
-      val instanceMirror = if (hasCompanion) Some(mirror reflect (mirror reflectModule companionObject.get).instance) else None
-      val companionMembers = if (hasCompanion) Some(instanceMirror.get.symbol.typeSignature) else None
-
-      constructor.paramLists.head.zipWithIndex.foreach { case (param, index) =>
-        val name = param.name.toString
-        val paramType = param.typeSignature
-        if (paramType.typeArgs.size > 1) throw new CommandLineException(s"Parameter $name has multiple generic types and that is not currently supported")
-
-        val paramClass = ReflectionUtil.typeToClass(paramType)
-        val paramUnitType = if (paramType.typeArgs.size == 1) paramType.typeArgs.head else paramType
-        val paramUnitClass = ReflectionUtil.typeToClass(paramUnitType)
-        val paramPrimitiveClass = ReflectionUtil.ifPrimitiveThenWrapper(paramUnitClass)
-
-        val paramTypeName = paramUnitType match {
-          case ref: TypeRef => ref.sym.name.toString
-          case _ => paramType.typeSymbol.asClass.name.decodedName.toString
+    constructor.paramLists match {
+      case Nil | _ :: _ :: _ =>
```

Comment:
can you name the `_` just to make it clearer (I could see them at first).  How about `xs :: rest :: Nil`?